infocmdb/ci_type.go

```diff
@@ -143,3 +144,139 @@ func (c *Client) SetTypeOfCi(ciId int, ciType string) (err error) {
 
 	return
 }
+
+type respCreateCiType struct {
+	Success bool         `json:"success"`
+	Message string       `json:"message"`
+	Data    []responseId `json:"data"`
+}
+
+type CiTypeParams struct {
+	Name                    string
+	Description             string
+	Note                    string
+	ParentCiTypeId          int
+	OrderNumber             int
+	createButtonDescription string
+	icon                    string
+	query                   string
+	DefaultProjectId        int
+	defaultAttributeId      int
+	defaultSortAttributeId  int
+	isDefaultSortAsc        int
+	IsCiAttach              int
+	IsAttributeAttach       int
+	tag                     string
+	isTabEnabled            int
+	isEventEnabled          int
+	IsActive                int
+	userId                  int
+}
+
+func (c *Client) NewCiTypeParams() (params *CiTypeParams) {
+	params = &CiTypeParams{
+		Name:                    "",
+		Description:             "",
+		Note:                    "",
+		ParentCiTypeId:          0,
+		OrderNumber:             0,
+		createButtonDescription: "",
+		icon:                    "",
+		query:                   "",
+		DefaultProjectId:        0,
+		defaultAttributeId:      0,
+		defaultSortAttributeId:  0,
+		isDefaultSortAsc:        0,
+		IsCiAttach:              0,
+		IsAttributeAttach:       0,
+		tag:                     "",
+		isTabEnabled:            0,
+		isEventEnabled:          0,
+		IsActive:                1,
+		userId:                  0,
+	}
+	return
+}
+
+func (c *Client) CreateCiType(typeParams *CiTypeParams) (typeId int, err error) {
+
+	if err = c.v2.Login(); err != nil {
+		return
+	}
+
+	typeIdExists, _ := c.GetCiTypeIdByCiTypeName(typeParams.Name)
```

Comment:
This likely works when the error is `ErrNoResult` (typeIdExists will use the default 0 value), but will likely fail otherwise.
We should rather check for errors here, there might be other problems like missing permissions, infoCMDB offline, ...
in which case we should just propagate it (return the error from `GetCiTypeIdByCiTypeName`).

Somewhat unrelated, the variable name `typeIdExists` sounds to me like a bool variable. Personally I would have named it `existingTypeId`, but I don't care too much.

Reply:
Changed the variable  name and i check for errors now. I ignore the error if the error is ErrNoResult, and then go on to create a new CI Type.